File: MovieShopEFCore/MovieShop.Web/Models/Movie.cs
namespace MovieShopMVC.Models;

public class Movie
{
    public int Id { get; set; }
    public string Title { get; set; } = "";
    public string PosterUrl { get; set; } = "";
    public string BackdropUrl { get; set; } = "";
    public string Overview { get; set; } = "";
    public string Tagline { get; set; } = "";
    public double Rating { get; set; }
    public string Genre { get; set; } = "";
    public List<string> Genres { get; set; } = [];
    public DateTime ReleaseDate { get; set; }
    public int Runtime { get; set; }
    public decimal Budget { get; set; }
    public decimal Revenue { get; set; }
    public decimal Price { get; set; }
    public List<Cast> Casts { get; set; } = [];
    public List<Trailer> Trailers { get; set; } = [];
    public List<Review> Reviews { get; set; } = [];
}

File: MovieShopAngular/src/app/movies/movie.ts
export interface MovieSummary {
  id: number;
  title: string;
  releaseDate: string;
  price: number;
  posterUrl: string;
  revenue: number;
}

export interface PagedResult<T> {
  items: T[];
  pageNumber: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;
  hasPreviousPage: boolean;
  hasNextPage: boolean;
}

export interface MovieCast { name: string; character: string; profileUrl: string | null; }
export interface MovieTrailer { name: string; url: string; }

export interface MovieDetails extends MovieSummary {
  backdropUrl: string;
  overview: string;
  tagline: string;
  rating: number;
  genre: string;
  genres: string[];
  runtime: number;
  budget: number;
  casts: MovieCast[];
  trailers: MovieTrailer[];
}
